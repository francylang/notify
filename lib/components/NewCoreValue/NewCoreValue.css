.new-value {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: fixed;
  top: 50%;
  left: 50%;
  z-index: 100;
  transform: translate(-50%, -50%);
  background-color: #fff;
  height: 270px;
  width: 500px;
  border-radius: 5px;
  padding: 30px;
  margin: 40px 0 0;
  box-shadow: 5px 10px 10px #bdbdbd; }

.create-new-value {
  text-align: center;
  color: #1a90bf;
  font-size: 35px; }

.add-new-value-form input {
  margin-top: 30px; }

.new-core-value-input {
  width: 100%;
  background-color: transparent;
  border: none;
  border-bottom: 1px solid #bdbdbd;
  font-size: 20px;
  font-weight: lighter;
  margin: 6px 0; }

.new-core-value-input:focus {
  outline: none;
  border-bottom: 2px solid #1a90bf; }

.form-button-value {
  background-color: #1a90bf;
  border: none;
  border-radius: 10px;
  color: #fff;
  padding: 10px 20px;
  font-size: 25px;
  margin: 20px 0;
  width: 100%;
  cursor: pointer; }

.form-button-value:hover {
  background-color: #405780; }

.form-button-value:disabled {
  background-color: #bdbdbd; }
