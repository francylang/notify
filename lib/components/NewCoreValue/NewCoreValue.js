import React, { Component } from 'react';
import PropTypes from 'prop-types';

class NewCoreValue extends Component {
  constructor(props) {
    super(props);
    this.state = {
      coreValue: ''
    };
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event, type) {
    let value = event.target.value;

    this.setState({
      [type]: value
    });
  }

  handleSubmit(event) {
    const { addCoreValue, history } = this.props;
    event.preventDefault();
    addCoreValue(this.state.coreValue);
    this.setState({
      coreValue: ''
    });
    history.push('/');
  }

  render() {
    const { coreValue } = this.state;
    return (
      <div className='new-value'>
        <h2 className='create-new-value'>Add New Core Value</h2>
          <form
            className='add-new-value-form'>
            <input
              type='text'
              placeholder='Core Value'
              className='new-core-value-input'
              value={coreValue}
              onChange={ (event) => this.handleChange(event, 'coreValue')}/>
            <input
              disabled={ !coreValue.length }
              className='form-button-value'
              type='submit'
              onClick={this.handleSubmit}/>
          </form>
      </div>
    );
  }
}

NewCoreValue.propTypes = {
  addCoreValue: PropTypes.func,
  isAuthenticated: PropTypes.func,
  history: PropTypes.object
};


export default NewCoreValue;
