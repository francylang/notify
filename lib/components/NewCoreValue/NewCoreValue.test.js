import React from 'react';
import NewCoreValue from './NewCoreValue.js';
import { shallow } from 'enzyme';

describe('<NewCoreValue />', () => {

  it('should exist', () => {
    const wrapper = shallow(<NewCoreValue />);

    expect(wrapper).toBeDefined();
  });

  it('should match the NewCoreValue snapshot', () => {
    const wrapper = shallow(<NewCoreValue />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should change state on input', () => {
    const wrapper = shallow(<NewCoreValue />);
    const coreValue = wrapper.find('.new-core-value-input');

    coreValue.simulate('change', { target: { value: 'Fun' } });
    expect(wrapper.state().coreValue).toEqual('Fun');
  });
});
