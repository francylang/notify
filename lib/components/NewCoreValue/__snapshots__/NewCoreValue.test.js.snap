// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewCoreValue /> should match the NewCoreValue snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="new-value"
>
    <h2
        className="create-new-value"
    >
        Add New Core Value
    </h2>
    <form
        className="add-new-value-form"
    >
        <input
            className="new-core-value-input"
            onChange={[Function]}
            placeholder="Core Value"
            type="text"
            value=""
        />
        <input
            className="form-button-value"
            disabled={true}
            onClick={[Function]}
            type="submit"
        />
    </form>
</div>,
  "nodes": Array [
    <div
      className="new-value"
>
      <h2
            className="create-new-value"
      >
            Add New Core Value
      </h2>
      <form
            className="add-new-value-form"
      >
            <input
                  className="new-core-value-input"
                  onChange={[Function]}
                  placeholder="Core Value"
                  type="text"
                  value=""
            />
            <input
                  className="form-button-value"
                  disabled={true}
                  onClick={[Function]}
                  type="submit"
            />
      </form>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <NewCoreValue />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": NewCoreValue {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleSubmit": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "coreValue": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="new-value"
>
          <h2
                    className="create-new-value"
          >
                    Add New Core Value
          </h2>
          <form
                    className="add-new-value-form"
          >
                    <input
                              className="new-core-value-input"
                              onChange={[Function]}
                              placeholder="Core Value"
                              type="text"
                              value=""
                    />
                    <input
                              className="form-button-value"
                              disabled={true}
                              onClick={[Function]}
                              type="submit"
                    />
          </form>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="new-value"
>
          <h2
                    className="create-new-value"
          >
                    Add New Core Value
          </h2>
          <form
                    className="add-new-value-form"
          >
                    <input
                              className="new-core-value-input"
                              onChange={[Function]}
                              placeholder="Core Value"
                              type="text"
                              value=""
                    />
                    <input
                              className="form-button-value"
                              disabled={true}
                              onClick={[Function]}
                              type="submit"
                    />
          </form>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <NewCoreValue />,
}
`;
